### PR TITLE
HIVE-27916: Increase tez.am.resource.memory.mb for TestIcebergCliDrver to 512MB

### PR DESCRIPTION
**Pull request by abstractdog**
### What changes were proposed in this pull request?
Increasing tez.am.resource.memory.mb for tez based iceberg cli driver.

### Why are the changes needed?
To improve test stability.

### Does this PR introduce _any_ user-facing change?
No.

### Is the change a dependency upgrade?
No.

### How was this patch tested?
Precommit.

### CODE REVIEW

File: data/conf/iceberg/tez/tez-site.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
   </property>
   <property>
     <name>tez.am.resource.memory.mb</name>
-    <value>256</value>
+    <value>512</value>
```
[the remark this answers is not in the document]
Comment (deniskuzZ) [reply data/conf/iceberg/tez/tez-site.xml:8]: btw, should we drop `hive.tez.container.size` from tez and llap tez-site.xml?

**Review by aturoczy — APPROVED** [review]

File: data/conf/iceberg/tez/tez-site.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (abstractdog) [reply data/conf/iceberg/tez/tez-site.xml:8]: hive.tez.container.size is for task containers in particular (it's used to create the Resource object which is provided to yarn), whereas tez.am.resource.memory.mb is for the Application Master

**Comment by abstractdog** [comment]
closing this, please find comment in jira: https://issues.apache.org/jira/browse/HIVE-27916?focusedCommentId=17796287&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17796287

**Comment by abstractdog** [comment]
other confusing stuff regarding hive.tez.container.size will be handled in HIVE-27937

**Comment by abstractdog** [comment]
looks like we need this change: downstream I was able to reproduce AM OOM flakiness which disappeared with this fix

**Pull request merged**